scripts/ci/update-changelog.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash -eux
+
+
+dependencies="cargo git git-cliff"
+
+for dependency in $dependencies
+do
+  if ! command -v $dependency &>/dev/null
+  then
+    echo "error: $dependency was not found in PATH" >&2
+    exit 255
+  fi
+done
+
+# The default "starting" commit is a somewhat arbitrary starting point for
+# cataloging recent commits in a way that breaks from the old convention
+DEFAULT_START_COMMIT="891e7d47827e99947e46c82f509e479aa13acf24"
+DEFAULT_NEWEST_COMMIT="$(git rev-parse HEAD)"
+DEFAULT_COMMIT_RANGE="${DEFAULT_START_COMMIT}..${DEFAULT_NEWEST_COMMIT}"
+DEFAULT_OUTPUT_FILE="CHANGELOG-UPDATED.md"
```

Comment:
Should this be `CHANGELOG.md`, to be consistent with the argument used in the CI job?

Reply:
I could probably call this out better in the help text, but the default output filename is different so that if you're running it locally you don't overwrite your original CHANGELOG.md without explicitly telling the script to do so. This is sort of a judgement call on what default behaviors to set for the script in light of some peculiar behaviors of `git-cliff` (such as e.g. only being able to use the native `--prepend` option if the `-l` or `-u` flags are passed, which we don't want to do in our use case).

Now that I've looked at it again, I think I'll restructure this to do away with setting the `-o` git-cliff option by default, which should mean that this variable will go away.